### PR TITLE
Fix probing of hwid

### PR DESCRIPTION
**Pull request by doanac**
We already know the value of the platform when this is built. OE
has it defined as "MACHINE", so this can just be set at build time
to eliminate the need to provide this value.

Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
This should make things a little easier for us. One thing to note - We *must* make an update to meta-lmp to pass `-DHARDWARE_ID=${MACHINE}`. Let me know if this seems reasonable.

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,11 @@ if(DOCKER_APPS)
     message(STATUS "Enabling docker-apps support")
 endif(DOCKER_APPS)
 
+IF (NOT DEFINED HARDWARE_ID)
+    message(FATAL_ERROR "Missing required value: HARDWARE_ID")
+ENDIF()
+ADD_DEFINITIONS(-DHARDWARE_ID="${HARDWARE_ID}")
```
Comment (mike-scott) [inline CMakeLists.txt]: Did this need to be ${MACHINE} (and the line underneath)

**Comment by mike-scott** [comment]
> This should make things a little easier for us. One thing to note - We *must* make an update to meta-lmp to pass `-DHARDWARE_ID=${MACHINE}`. Let me know if this seems reasonable.

Heh, we were cross-commenting.  Yes, I think this is reasonable.

**Comment by ricardosalveti** [comment]
Yeah, OE can set HARDWARE_ID to MACHINE.

**Pull request merged**